src/class/SimpleData.ts

```diff
@@ -66,378 +87,425 @@ export default class SimpleData {
+            nbDigits,
+            nbValuesTestedForTypeOf,
+            key1,
+            key2,
         )
-        this.#updateSimpleData(data)
+        overwrite && this.#updateSimpleData(data)
+
         return this
     }
 
-    renameKey(...args: any[]) {
-        const data = this.#apply(
-            renameKey_,
-            ...args
+    @logCall()
+    excludeMissingValues({ 
+        key, 
+        missingValues, 
+        overwrite = true 
+    }: { 
+        key?: string, 
+        missingValues?: any[], 
+        overwrite?: boolean 
+    } = {}): SimpleData {
+        if (missingValues === undefined) {
+            missingValues = [null, NaN, undefined, ""]
+        }
+        const data = excludeMissingValues_(
+            this.data,
+            key,
+            missingValues,
+            this._verbose
         )
-        this.#updateSimpleData(data)
+        overwrite && this.#updateSimpleData(data)
+
         return this
     }
 
-    removeKey(...args: any[]) {
-        const data = this.#apply(
-            removeKey_,
-            ...args
-        )
-        this.#updateSimpleData(data)
+    @logCall()
+    renameKey({ oldKey, newKey, overwrite = true }: { oldKey: string, newKey: string, overwrite?: boolean }): SimpleData {
+        const data = renameKey_(this._data, oldKey, newKey)
+        overwrite && this.#updateSimpleData(data)
+
         return this
     }
 
-    addKey(...args: any[]) {
-        const data = this.#apply(
-            addKey_,
-            ...args
-        )
-        this.#updateSimpleData(data)
+    @logCall()
+    removeKey({ key, overwrite = true }: { key: string, overwrite?: boolean }): SimpleData {
+        const data = removeKey_(this._data, key)
+        overwrite && this.#updateSimpleData(data)
+
         return this
     }
 
-    selectKeys(...args: any[]) {
-        const data = this.#apply(
-            selectKeys_,
-            ...args
-        )
-        this.#updateSimpleData(data)
+    @logCall()
+    addKey({ key, func, overwrite = true }: { key: string, func: (item: any) => any, overwrite?: boolean }): SimpleData {
+        const data = addKey_(this._data, key, func)
+        overwrite && this.#updateSimpleData(data)
+
         return this
     }
 
-    modifyValues(...args: any[]) {
-        const data = this.#apply(
-            modifyValues_,
-            ...args
-        )
-        this.#updateSimpleData(data)
+    @logCall()
+    selectKeys({ keys, overwrite = true }: { keys: string[], overwrite?: boolean }): SimpleData {
+        const data = selectKeys_(this._data, keys)
+        overwrite && this.#updateSimpleData(data)
+
         return this
     }
 
-    modifyItems(...args: any[]) {
-        const data = this.#apply(
-            modifyItems_,
-            ...args
-        )
-        this.#updateSimpleData(data)
+    @logCall()
+    modifyValues({ key, func, overwrite = true }: { key: string, func: (val: any) => any, overwrite?: boolean }): SimpleData {
```

Comment:
For the (val: any), maybe we could create a type simpleDataValue in SimpleData.types.ts ?